### PR TITLE
[Backend] Tag `source_file_changes` with the type of change given by `git log`

### PR DESCRIPTION
**Pull request by zergov**
Closes https://github.com/visevol/Cscope/issues/94

Categorize type of change in RepositorySyncService

Update the RepositorySyncService to set the `category` column on each
`source_file_changes` rows it creates to the type of change returned by
`git log`.

Change that creates a file are tagged with the `create` category.
Change that modifies a file are tagged with the `modify` category.
Change that deletes a file are tagged with the `delete` category.

### CODE REVIEW

File: api/app/models/source_file_change.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (zergov) [inline api/app/models/source_file_change.rb:1]: Context on this comment: https://blog.saeloun.com/2024/05/20/frozen-string-literal/

TL;DR it freezes string literals, making them immutable. I added this so that the constants for the change category cannot be modified. 

Normally this comment should be added to all files, but for some reason I didn't do it 🤷

**Comment by zergov** [comment]
It looks like when we query on the ledger of changes, it's fine:
```ruby
Repository.first.source_files.find_by(filepath: "Rakefile").commits.on_changes_ledger
> 1
```
